### PR TITLE
APIView Style

### PR DESCRIPTION
**Pull request by ricardofsc10**
GET method work 100%
DELETE, POST methods not tested
Models.py 100% correct and working

APIViews style correct and created for every view

### CODE REVIEW

File: tuichain/api/models.py [hunk 1]
```diff
@@ -11,16 +11,23 @@ class Student(models.Model):
...
+         return self.full_name
 
 
 class LoanRequest(models.Model):
-    request_date = models.DateTimeField()
+    request_date = models.DateTimeField(auto_now_add=True)
     school = models.CharField(max_length=100)
     course = models.CharField(max_length=100)
     amount = models.DecimalField(max_digits=8, decimal_places=2)
+    current_amount = models.DecimalField(editable=False, max_digits=8, decimal_places=2, default='0.00')
```
Comment (hpereira98) [inline tuichain/api/models.py:25]: editable cannot be false here

File: tuichain/api/models.py [hunk 2]
```diff
@@ -11,16 +11,23 @@ class Student(models.Model):
...
     student = models.ForeignKey(Student, on_delete=models.CASCADE)
-    validated = models.BooleanField()
+    validated = models.BooleanField(editable=False,default=False)
```
Comment (hpereira98) [inline tuichain/api/models.py:27]: same as above

File: tuichain/api/serializers.py [hunk 3]
```diff
@@ -4,19 +4,59 @@
...
+        return Investor.objects.create(**validated_data)
 
 class LoanRequestSerializer(serializers.ModelSerializer):
     class Meta:
         model = LoanRequest
-        fields = '__all__'
+        fields = ('school',
+                'course',
+                'amount',
+                'student')
```
Comment (hpereira98) [inline tuichain/api/serializers.py:40]: why not pass the all the fields here?

**Review by hpereira98 — CHANGES_REQUESTED** [review]

**Review by ricardomilhazes — APPROVED** [review]
Seems good
Views are way more understandable now

File: tuichain/api/serializers.py (hunk 3 above)
Comment (ricardomilhazes) [reply tuichain/api/serializers.py:40]: Guess it's because 'creation_date' is not important

Comment (ricardomilhazes) [reply tuichain/api/serializers.py:40]: But it might be in some cases, not sure

Comment (mrr37) [reply tuichain/api/serializers.py:40]: and also current amount right?

**Review by mrr37 — APPROVED** [review]
Loanrequest serializer might need some additions, namely request_date and current_amount

**Pull request closed without merging**